### PR TITLE
chore: Update pre-commit to run prettier

### PR DESCRIPTION
**Pull request by SamoKopecky**
Update `pre-commit` to run prettier, run prettier on all files. Prettier should use `@spotify/prettier-config` since it is configured in the root `package.json` file, see [configuration docs](https://prettier.io/docs/en/configuration.html).

/cc @tumido

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -25,3 +25,8 @@ repos:
         files: \.(yaml|yml)$
         types: [file, yaml]
         entry: yamllint --strict -c yamllint-config.yaml
+
+  - repo: https://github.com/pre-commit/mirrors-prettier
+    rev: v2.7.1
+    hooks:
+      - id: prettier
```
Comment (tumido) [inline .pre-commit-config.yaml:32]: This is not using the configuration you've mentioned. Please read the pre-commit `prettier` hook docs and see the CI logs:
image

**Review by tumido — CHANGES_REQUESTED** [review]

File: .pre-commit-config.yaml (hunk 1 above)
Comment (SamoKopecky) [reply .pre-commit-config.yaml:32]: Totally missed that, sorry about that working on fixing it.

**Review by tumido — APPROVED** [review]
/lgtm
/approve

**Comment by SamoKopecky** [comment]
`pre-commit` failed cause the PR that just got merged had changes that didn't comply with prettier. I rebased to `main` so it should finally work now.

**Review by tumido — APPROVED** [review]
/lgtm

**Pull request merged**